### PR TITLE
refactor(autoware_planning_evaluator): devops node dojo

### PR DESCRIPTION
**Pull request by odra**
## Description

Implement ROS Node configuration according to the devops dojo instruction document for the "autoware_planning_evaluator" package.

## Related links

* https://github.com/orgs/autowarefoundation/discussions/3371

**Parent Issue:**

-  https://github.com/orgs/autowarefoundation/discussions/3371

## How was this PR tested?

```
colcon build --symlink-install --packages-up-to autoware_planning_evaluator
```

```
ros2 launch evaluator/autoware_planning_evaluator/launch/planning_evaluator.launch.xml
```

## Notes for reviewers

None.

## Interface changes

No interface changes per say, just moving the existing param files to the "config" folder and creating a json schema file containing that same information.

## Effects on system behavior

None.

### CODE REVIEW

**Review by maxime-clem — CHANGES_REQUESTED** [review]
The tests are failing because the path to the parameter file is not updated (https://github.com/autowarefoundation/autoware.universe/actions/runs/10692393280/job/29640731369?pr=8746).
Please also fix some typos such that the spell check passes (https://github.com/autowarefoundation/autoware.universe/actions/runs/10688815334/job/29640724521?pr=8746).

Finally the README fails to render: https://autowarefoundation.github.io/autoware.universe/pr-8746/evaluator/autoware_planning_evaluator/
```
FileNotFoundError: [Errno 2] No such file or directory: 'validator/autoware_planning_evaluator/schema/autoware_planning_evaluator.schema.json'
```

Otherwise the PR looks good and I will approve it once the above issues are fixed.

**Comment by odra** [comment]
@maxime-clem could you review it again? I believe I've  applied the required changes.

File: evaluator/autoware_planning_evaluator/schema/autoware_planning_evaluator.schema.json [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+            "stability_frechet",
+            "obstacle_distance",
+            "obstacle_ttc",
+            "modified_goal_longitudinal_deviation",
+            "modified_goal_lateral_deviation",
+            "modified_goal_yaw_deviation"
+          ]
+        },
+        "trajectory": {
+          "description": "trajectory object",
```
Comment (maxime-clem) [inline evaluator/autoware_planning_evaluator/schema/autoware_planning_evaluator.schema.json:47]: I think the [documentation rendering issue](https://autowarefoundation.github.io/autoware.universe/pr-8746/evaluator/autoware_planning_evaluator/) is caused by missing `type` fields.
```suggestion
          "description": "trajectory object",
          "type": "object",
```

File: evaluator/autoware_planning_evaluator/schema/autoware_planning_evaluator.schema.json [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+          "properties": {
+            "min_point_dist_m": {
+              "description": "minimum distance between two successive points to use for angle calculation",
+              "type": "number",
+              "default": 0.1
+            },
+            "lookahead": {
+              "description": "lookahead object",
```
Comment (maxime-clem) [inline evaluator/autoware_planning_evaluator/schema/autoware_planning_evaluator.schema.json:55]:
```suggestion
              "description": "lookahead object",
              "type": "object",
```

**Review by maxime-clem — COMMENTED** [review]
Thank you for the changes. I think there is only one issue left to fix and then we can merge the PR.

File: evaluator/autoware_planning_evaluator/schema/autoware_planning_evaluator.schema.json (hunk 2 above)
Comment (odra) [reply evaluator/autoware_planning_evaluator/schema/autoware_planning_evaluator.schema.json:56]: Added in both now.

**Review by maxime-clem — APPROVED** [review]

**Pull request merged**